src/org/daisy/dotify/tasks/impl/input/xml/resource-files/common/xslt-files/dtbook2obfl_braille.xsl

```diff
@@ -267,20 +267,28 @@
 	
 	<xsl:template match="dtb:level1|dtb:level2" mode="toc">
 		<xsl:if test="dtb:h1|dtb:h2">
-<!--
-			<xsl:choose>
-				<xsl:when test="self::dtb:level1 and @class='part'"><toc-entry ref-id="{generate-id(.)}" padding-bottom="1" keep="page"><xsl:apply-templates select="dtb:h1" mode="toc-hd"/></toc-entry><xsl:apply-templates mode="toc"/></xsl:when>
-				<xsl:otherwise>--><toc-entry ref-id="{generate-id(dtb:h1|dtb:h2)}" block-indent="{$toc-indent-multiplier}" text-indent="{2*$toc-indent-multiplier}" keep="page"><xsl:apply-templates select="dtb:h1|dtb:h2" mode="toc-hd"/><xsl:apply-templates mode="toc"/></toc-entry><!--</xsl:otherwise>
-			</xsl:choose>-->
+			<toc-block block-indent="{$toc-indent-multiplier}" text-indent="{2*$toc-indent-multiplier}" keep="page">
+				<toc-entry ref-id="{generate-id(dtb:h1|dtb:h2)}" >
+					<xsl:apply-templates select="dtb:h1|dtb:h2" mode="toc-hd"/>
+				</toc-entry>
+				<xsl:apply-templates mode="toc"/>
+			</toc-block>
 		</xsl:if>
 	</xsl:template>
 	
 	<xsl:template match="dtb:level3|dtb:level4|dtb:level5|dtb:level6" mode="toc">
 		<xsl:if test="dtb:h3|dtb:h4|dtb:h5|dtb:h6">
-			<toc-entry ref-id="{generate-id(dtb:h3|dtb:h4|dtb:h5|dtb:h6)}" block-indent="{$toc-indent-multiplier}" text-indent="{$toc-indent-multiplier}" keep="page"><xsl:apply-templates select="dtb:h3|dtb:h4|dtb:h5|dtb:h6" mode="toc-hd"/>
-			<xsl:if test="dtb:level3 and ancestor::dtb:level1[@class='part']"><xsl:apply-templates mode="toc"/></xsl:if>
-			</toc-entry>
-			<xsl:if test="not(dtb:level3 and ancestor::dtb:level1[@class='part'])"><xsl:apply-templates mode="toc"/></xsl:if>
+			<toc-block block-indent="{$toc-indent-multiplier}" text-indent="{$toc-indent-multiplier}" keep="page">
```

Comment:
The `toc-block` has no `keep` attribute. We decided not to use the special attributes

Reply:
Ok, so i should remove the `keep` attribute from from toc-blocks?

Reply:
@bertfrees where should the `keep` attribute but put? The only element that has a `keep` attribute is `block` but there is no such thing in the `table-of-contents`.

Reply:
> We decided not to use the special attributes

There has been some confusion. I indeed said somewhere in the discussion that `toc-entry` doesn't have the special attributes, but this is only the case in the documentation. In the RelaxNG schema `toc-entry` does have all the attributes actually. When we were discussing the attributes for `toc-block` we also included all the attributes in the discussion.